qemu/tests/x86_cpu_test_dies.py

```diff
@@ -0,0 +1,45 @@
+import random
+
+from avocado.utils import cpu
+
+from virttest import error_context
+from virttest import env_process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Check guest gets correct multiple vcpu dies
+
+    1) Boot guest with options: -smp n,dies=2x...
+    2) Check cpu dies(only for Linux guest and Intel host)
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    vm_name = params['main_vm']
+    vcpu_dies_list = [2, 4]
+    params['vcpu_dies'] = random.choice(vcpu_dies_list)
+    params['start_vm'] = 'yes'
+    env_process.preprocess_vm(test, params, env, vm_name)
+    vm = env.get_vm(vm_name)
+    session = vm.wait_for_login()
+    if params["os_type"] == "linux" and cpu.get_vendor() == 'intel':
+        check_die_id = params['check_die_id']
+        check_die_cpus_list = params['check_die_cpus_list']
+        vcpu_sockets = vm.cpuinfo.sockets
+        vcpu_dies = vm.cpuinfo.dies
+        dies_id = session.cmd_output(check_die_id).strip().split('\n')
+        dies_cpus_list = session.cmd_output(
+            check_die_cpus_list).strip().split('\n')
+        if len(dies_id) != int(vcpu_dies):
+            test.fail("die_id is not right: %d != %d"
+                      % (len(dies_id), int(vcpu_dies)))
+        if len(dies_cpus_list) != int(vcpu_sockets)*int(vcpu_dies):
```

Comment:
Just confirm, `uniq -c` has already count the output of the command, here still nee `len()`, it should always be `1` I think, is that what you want?

Reply:
> Just confirm, `uniq -c` has already count the output of the command, here still need `len()`? It should always be `1` I think, is that what you want?

I split the output with '\n' to count the `die_id` or `die_cpus_list`, this is the output:
```
cat /sys/devices/system/cpu/cpu*/topology/die_id | sort | uniq -c
     64 0
     64 1
```
there are two dies.

```
# cat /sys/devices/system/cpu/cpu*/topology/die_cpus_list | sort | uniq -c
     16 0-15
     16 112-127
     16 16-31
     16 32-47
     16 48-63
     16 64-79
     16 80-95
     16 96-111
```

Reply:
@nanliu-r Thanks for the clarification, it seems clear and works in right way.